File: OPENAPI-REST-API/swagger-client/dart/lib/model/inline_response20012_filter_firewall_rules.dart
part of swagger.api;

class InlineResponse20012FilterFirewallRules {
  
  String id = null;

  String sourceIp = null;

  String destinationIp = null;

  String protocolId = null;

  String sourcePort = null;

  String destinationPort = null;

  String xdpAction = null;

  String globalDrop = null;

  InlineResponse20012FilterFirewallRules();

  @override
  String toString() {
    return 'InlineResponse20012FilterFirewallRules[id=$id, sourceIp=$sourceIp, destinationIp=$destinationIp, protocolId=$protocolId, sourcePort=$sourcePort, destinationPort=$destinationPort, xdpAction=$xdpAction, globalDrop=$globalDrop, ]';
  }

  InlineResponse20012FilterFirewallRules.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    id = json['id'];
    sourceIp = json['source_ip'];
    destinationIp = json['destination_ip'];
    protocolId = json['protocol_id'];
    sourcePort = json['source_port'];
    destinationPort = json['destination_port'];
    xdpAction = json['xdp_action'];
    globalDrop = json['global_drop'];
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'source_ip': sourceIp,
      'destination_ip': destinationIp,
      'protocol_id': protocolId,
      'source_port': sourcePort,
      'destination_port': destinationPort,
      'xdp_action': xdpAction,
      'global_drop': globalDrop
     };
  }

  static List<InlineResponse20012FilterFirewallRules> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20012FilterFirewallRules>() : json.map((value) => new InlineResponse20012FilterFirewallRules.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20012FilterFirewallRules> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, InlineResponse20012FilterFirewallRules>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new InlineResponse20012FilterFirewallRules.fromJson(value));
    }
    return map;
  }
}

File: OPENAPI-REST-API/swagger-client/csharp-dotnet2/src/main/CsharpDotNet2/myadmin-client-csharp-dotnet2/Model/InlineResponse20012FilterFirewallRules.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class InlineResponse20012FilterFirewallRules {
    /// <summary>
    /// Gets or Sets Id
    /// </summary>
    [DataMember(Name="id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "id")]
    public string Id { get; set; }

    /// <summary>
    /// Gets or Sets SourceIp
    /// </summary>
    [DataMember(Name="source_ip", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "source_ip")]
    public string SourceIp { get; set; }

    /// <summary>
    /// Gets or Sets DestinationIp
    /// </summary>
    [DataMember(Name="destination_ip", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "destination_ip")]
    public string DestinationIp { get; set; }

    /// <summary>
    /// Gets or Sets ProtocolId
    /// </summary>
    [DataMember(Name="protocol_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "protocol_id")]
    public string ProtocolId { get; set; }

    /// <summary>
    /// Gets or Sets SourcePort
    /// </summary>
    [DataMember(Name="source_port", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "source_port")]
    public string SourcePort { get; set; }

    /// <summary>
    /// Gets or Sets DestinationPort
    /// </summary>
    [DataMember(Name="destination_port", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "destination_port")]
    public string DestinationPort { get; set; }

    /// <summary>
    /// Gets or Sets XdpAction
    /// </summary>
    [DataMember(Name="xdp_action", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "xdp_action")]
    public string XdpAction { get; set; }

    /// <summary>
    /// Gets or Sets GlobalDrop
    /// </summary>
    [DataMember(Name="global_drop", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "global_drop")]
    public string GlobalDrop { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class InlineResponse20012FilterFirewallRules {\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("  SourceIp: ").Append(SourceIp).Append("\n");
      sb.Append("  DestinationIp: ").Append(DestinationIp).Append("\n");
      sb.Append("  ProtocolId: ").Append(ProtocolId).Append("\n");
      sb.Append("  SourcePort: ").Append(SourcePort).Append("\n");
      sb.Append("  DestinationPort: ").Append(DestinationPort).Append("\n");
      sb.Append("  XdpAction: ").Append(XdpAction).Append("\n");
      sb.Append("  GlobalDrop: ").Append(GlobalDrop).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
